src/Promotions/InPluginUpsells/resources/js/donation-options.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ *
+ * @unreleased
+ *
+ */
+declare global {
+    interface Window {
+        GiveLegacyFormEditor: GiveLegacyFormEditorData;
+    }
+}
+
+export interface GiveLegacyFormEditorData {
+    apiRoot: string;
+    apiNonce: string;
+}
+
+/**
+ *
+ * @unreleased
+ *
+ */
+const recurringProductRecommendation = document.querySelector('.givewp-donation-options');
+const dismissAction = document.querySelector('.givewp-donation-options_close');
+const preceedingContent = document.querySelector('._give_custom_amount_text_field');
+
+if (recurringProductRecommendation && preceedingContent) {
+    preceedingContent.insertAdjacentElement('afterend', recurringProductRecommendation);
+
+    dismissAction.addEventListener('click', async function (event) {
+        recurringProductRecommendation.remove();
+        await postRequest();
+    });
+}
+
+/**
+ *
+ * @unreleased
+ *
+ */
+async function postRequest() {
```

Comment:
This function name can be updated to be more descriptive. While it does post a request the name lacks domain/business context.

Reply:
How do we feel about this? https://github.com/impress-org/givewp/pull/6802/commits/2df3085d269bf7d653dac8f0cfa3227fbcd75caf ?